File: test_osrm.js
const fetch = require('node-fetch');
const kandigaiBins = require('./constants/kandigaiData').KANDIGAI_BINS;

async function test() {
    const driverLocation = { lat: 12.827, lng: 80.219 }; // SSV Branch 3 Siruseri approx
    const dumpYard = { lat: 12.9229, lng: 80.1274 }; // Tambaram MSW

    let sortedBins = [...kandigaiBins].sort((a, b) => {
        const numA = parseInt(a.id.replace(/\D/g, '')) || 0;
        const numB = parseInt(b.id.replace(/\D/g, '')) || 0;
        return numA - numB;
    });

    const coords = [
        [driverLocation.lng, driverLocation.lat],
        ...sortedBins.map(b => [b.coordinates.lng, b.coordinates.lat]),
        [dumpYard.lng, dumpYard.lat]
    ];

    const url = `https://router.project-osrm.org/route/v1/driving/${coords.map(c => c.join(',')).join(';')}?overview=full&geometries=geojson`;
    console.log("URL Length:", url.length);
    try {
        const res = await fetch(url);
        const data = await res.json();
        if (data.routes && data.routes[0]) {
            console.log("Distance:", data.routes[0].distance / 1000, "km");
            console.log("Duration:", data.routes[0].duration / 60, "min");
        } else {
            console.log("Error:", data);
        }
    } catch (e) {
        console.error(e);
    }
}
test();

File: test_osrm.py
import urllib.request
import json
import sys

coords = [
    [80.15399, 12.835573],
    [80.144675, 12.855117],
    [80.150373, 12.848659],
    [80.14901, 12.845991],
    [80.147045, 12.851268]
]

coord_str = ";".join([f"{c[0]},{c[1]}" for c in coords])
url = f"https://router.project-osrm.org/trip/v1/driving/{coord_str}?roundtrip=false&source=first&destination=last&overview=full&geometries=geojson"

try:
    req = urllib.request.Request(url)
    with urllib.request.urlopen(req) as response:
        data = json.loads(response.read().decode())
        if 'trips' in data:
            print("Trip successful!")
            print("Distance:", data['trips'][0]['distance'])
            print("Waypoints mapped:", len(data['waypoints']))
            # to get the new order, waypoints have a waypoint_index
            for wp in data['waypoints']:
                print(wp['waypoint_index'], wp['location'])
        else:
            print("No trips found in response")
except Exception as e:
    print("Error:", e)
